lib/vault/client.rb

```diff
@@ -177,8 +178,8 @@ def request(verb, path, data = {}, headers = {})
       # Create the HTTP connection object - since the proxy information defaults
       # to +nil+, we can just pass it to the initializer method instead of doing
       # crazy strange conditionals.
-      connection = Net::HTTP.new(uri.host, uri.port,
-        proxy_address, proxy_port, proxy_username, proxy_password)
+      connection = Net::HTTP::Persistent.new(name: "vault-ruby")
```

Comment:
This seems incorrect. Where are we using the proxy information now? Where is the URI coming from?